lntest/itest/lnd_zero_conf_test.go

```diff
@@ -1068,3 +1070,209 @@ func testOptionScidUpgrade(net *lntest.NetworkHarness, t *harnessTest) {
 		},
 	)
 }
+
+// testZeroConfReorg tests that a reorg does not cause a zero-conf channel to
+// be deleted from the channel graph. This was previously the case due to logic
+// in the function DisconnectBlockAtHeight.
+func testZeroConfReorg(net *lntest.NetworkHarness, t *harnessTest) {
+
+	if net.BackendCfg.Name() == lntest.NeutrinoBackendName {
+		t.Skipf("skipping zero-conf reorg test for neutrino backend")
+	}
+
+	var (
+		ctxb = context.Background()
+		temp = "temp"
+	)
+
+	// Since zero-conf is opt in, the harness nodes provided won't be able
+	// to open zero-conf channels. In that case, we just spin up new nodes.
+	zeroConfArgs := []string{
+		"--protocol.option-scid-alias",
+		"--protocol.zero-conf",
+		"--protocol.anchors",
+	}
+
+	carol := net.NewNode(t.t, "Carol", zeroConfArgs)
+	defer shutdownAndAssert(net, t, carol)
+
+	// Spin-up Dave so Carol can open a zero-conf channel to him.
+	dave := net.NewNode(t.t, "Dave", zeroConfArgs)
+	defer shutdownAndAssert(net, t, dave)
+
+	// We'll give Carol some coins in order to fund the channel.
+	net.SendCoins(t.t, btcutil.SatoshiPerBitcoin, carol)
+
+	// Ensure that both Carol and Dave are connected.
+	net.EnsureConnected(t.t, carol, dave)
+
+	// Setup a ChannelAcceptor for Dave.
+	ctxc, cancel := context.WithCancel(ctxb)
+	acceptStream, err := dave.ChannelAcceptor(ctxc)
+	require.NoError(t.t, err)
+	go acceptChannel(t.t, true, acceptStream)
+
+	// Open a private zero-conf anchors channel of 1M satoshis.
+	params := lntest.OpenChannelParams{
+		Amt:            btcutil.Amount(1_000_000),
+		CommitmentType: lnrpc.CommitmentType_ANCHORS,
+		ZeroConf:       true,
+	}
+	chanOpenUpdate := openChannelStream(t, net, carol, dave, params)
+
+	// Remove the ChannelAcceptor.
+	cancel()
+
+	// We should receive the OpenStatusUpdate_ChanOpen update without
+	// having to mine any blocks.
+	fundingPoint, err := net.WaitForChannelOpen(chanOpenUpdate)
+	require.NoError(t.t, err, "error while waiting for channel open")
+
+	err = carol.WaitForNetworkChannelOpen(fundingPoint)
+	require.NoError(t.t, err, "carol didn't report channel")
+	err = dave.WaitForNetworkChannelOpen(fundingPoint)
+	require.NoError(t.t, err, "dave didn't report channel")
+
+	// Attempt to send a 10K satoshi payment from Carol to Dave. This
+	// requires that the edge exists in the graph.
+	daveInvoiceParams := &lnrpc.Invoice{
+		Value: int64(10_000),
+	}
+	daveInvoiceResp, err := dave.AddInvoice(ctxb, daveInvoiceParams)
+	require.NoError(t.t, err, "unable to add invoice")
+
+	_ = sendAndAssertSuccess(
+		t, carol, &routerrpc.SendPaymentRequest{
+			PaymentRequest: daveInvoiceResp.PaymentRequest,
+			TimeoutSeconds: 60,
+			FeeLimitMsat:   noFeeLimitMsat,
+		},
+	)
+
+	// We will now attempt to query for the alias SCID in Carol's graph.
+	// The starting alias is 16_000_000:0:0 so that is the SCID we will
```

Comment:
How do we know that this is the starting alias?

Reply:
It's what's defined in the aliasmgr. Though I think we can just import it instead